### PR TITLE
(CDAP-4766) Skip fork node names in system tags for workflows

### PR DESCRIPTION
**Pull request by bdmogal**
Jira: [CDAP-4766](https://issues.cask.co/browse/CDAP-4766)
Build: http://builds.cask.co/browse/CDAP-DUT3911-3

### CODE REVIEW

**Comment by bdmogal** [comment]
@poornachandra addressed comments. please review again.

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/deploy/pipeline/SystemMetadataWriterStageTest.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    SystemMetadataWriterStage systemMetadataWriterStage = new SystemMetadataWriterStage(metadataStore);
+    StageContext stageContext = new StageContext(Object.class);
+    systemMetadataWriterStage.process(stageContext);
+    systemMetadataWriterStage.process(appWithPrograms);
+
+    // verify that the workflow is not tagged with the fork node name
+    for (String tag : metadataStore.getTags(MetadataScope.SYSTEM, appId.workflow(workflowName).toId())) {
+      try {
+        //noinspection ResultOfMethodCallIgnored
+        Integer.parseInt(tag);
```
Comment (bdmogal) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/deploy/pipeline/SystemMetadataWriterStageTest.java]: Good point. Fixed by using fork nodes from workflow spec.

**Comment by poornachandra** [comment]
LGTM

**Comment by bdmogal** [comment]
Squashed commits. Waiting for a clean build.

**Comment by bdmogal** [comment]
Build 🍏 : http://builds.cask.co/browse/CDAP-DUT3911-6. Merging.

**Pull request merged**